contracts/modules/v2/StakingPool.sol

```diff
@@ -113,43 +109,38 @@ contract StakingPool is IStakingPool, ERC721 {
   uint public constant CAPACITY_REDUCTION_DENOMINATOR = 10_000;
   uint public constant INITIAL_PRICE_DENOMINATOR = 10_000;
 
-  // product params flags
-  uint constant FLAG_PRODUCT_WEIGHT = 1;
-  uint constant FLAG_PRODUCT_PRICE = 2;
-
-  // withdraw flags
-  uint constant FLAG_WITHDRAW_DEPOSIT = 1;
-  uint constant FLAG_WITHDRAW_REWARDS = 2;
-
   modifier onlyCoverContract {
-    // TODO: restrict calls to cover contract only
+    require(msg.sender == coverContract, "StakingPool: Only Cover contract can call this function");
     _;
   }
 
   modifier onlyManager {
-    require(msg.sender == manager, "StakingPool: Only pool manager can call this function");
+    require(_isApprovedOrOwner(msg.sender, 0), "StakingPool: Only pool manager can call this function");
```

Comment:
so this implies the management of the pool is handed over by transferring the NFT right?

Reply:
Yes, correct